docs/proposals/metadata.md

```diff
@@ -0,0 +1,80 @@
+# Image Metadata
+
+## Abstract
+
+A proposal for the additional metadata for Docker images which will provide more
+context about the content of the image and define relations between Docker
+images.
+
+## Motivation
+
+Image metadata are needed to do automatic generation of the OpenShift v3
+resource templates, like PodTemplates, ReplicationControllers, BuildConfigs
+etc.  Automatic generation of a PodTemplate is also required for
+Deployments.
+
+## Constraints and Assumptions
+
+For the time being, there is no good way to define metadata for Docker
+images other than specifying extra environment variables. There is upstream
+work currently being done to fix this situation and we should switch to that in
+the implementation in the future.
+
+This document only defines the metadata needed by the current set of use cases.
+Additional metadata and/or use cases may be added in the future.
+
+## Use cases
+
+1. As an author of Docker image that is going to be consumed by the OpenShift
+   platform, I want to express what categories/tags my Docker image will belongs
+   to, so the platform then can then consume the tags to improve the generation
+   workflow based on them.
+
+2. As an user of OpenShift I want to get reliable suggestions about services
+   that the Docker image I'm going to use might require to operate properly. As
+   an author of the Docker image, I need to have a way to record what services my
+   Docker image want to consume.
+
+2. As an author of the Docker image, I need to have a way to indicate whether
+   the container started from my Docker image does not support scaling.
+   The UI should then communicate this information with the end consumers.
+
+## Metadata variables (required for 3.0)
+
+### TAGS
```

Comment:
The prefix is probably going to have to be more unique - if we start colliding with values end users provide that's bad.

----- Original Message -----
> > +   platform, I want to express what categories/tags my Docker image will
> > belongs
> > +   to, so the platform then can then consume the tags to improve the
> > generation
> > +   workflow based on them.
> > +
> > +2. As an user of OpenShift I want to get reliable suggestions about
> > services
> > +   that the Docker image I'm going to use might require to operate
> > properly. As
> > +   an author of the Docker image, I need to have a way to record what
> > services my
> > +   Docker image want to consume.
> > +
> > +2. As an author of the Docker image, I need to have a way to indicate
> > whether
> > +   the container started from my Docker image does not support scaling.
> > +   The UI should then communicate this information with the end consumers.
> > +
> > +## Metadata variables (required for 3.0)
> > +
> > +### TAGS
> 
> @smarterclayton I like IMAGE_
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/906/files#r24571508